application-ckeditor-plugins/src/main/resources/xwiki-macro/macroEditor.js

```diff
@@ -44,7 +44,8 @@ define('macroService', ['jquery', 'xwiki-meta'], function($, xcontext) {
     } else {
       var url = new XWiki.Document('MacroService', 'CKEditor').getURL('get', $.param({
         outputSyntax: 'plain',
-        language: $('html').attr('lang')
+        language: $('html').attr('lang'),
+        sourceDocument: xcontext.documentReference.toString()
```

Comment:
The source document reference is not necessarily the current document. See https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-14.10/xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-syntax/plugin.js#L49 .